src/main/java/edu/harvard/iq/dataverse/dataverse/featured/DataverseFeaturedItem.java

```diff
@@ -85,4 +99,42 @@ public String getImageFileUrl() {
         }
         return null;
     }
+
+    public String getType() {
+        return type;
+    }
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public DvObject getDvObject() {
+        return dvobject;
+    }
+
+    public void setDvObject(DvObject dvObject) {
+        this.dvobject = dvObject;
+    }
+
+    /*
+    Make sure the object and type match.
+     */
+    public static DvObjectFeaturedItem sanitizeDvObject(String type, DvObject dvObject) {
+        String dvType = (type != null) ? type.toLowerCase() : "custom";
+        if (DataverseFeaturedItem.VALID_TYPES.contains(dvType)) {
+            if (("dataverse".equals(dvType) && dvObject instanceof Dataverse) ||
+                    ("dataset".equals(dvType) && dvObject instanceof Dataset) ||
+                    ("datafile".equals(dvType) && dvObject instanceof DataFile)) {
+                return new DvObjectFeaturedItem(dvType, dvObject);
```

Comment:
Additionally, I think it's important to notify the client when an invalid type is specified. Currently, if someone sends something like "foo", the code quietly falls back to "custom", which could mask bugs or misuse of the API. If we move this logic into the API layer as suggested, it would be straightforward to perform validation and return a 400 Bad Request when the type is invalid.